lab_2_tokenize_by_bpe/main.py

```diff
@@ -81,6 +188,22 @@ def decode(
     :param end_of_word_token: an end-of-word token
     :return: decoded sequence
     """
+    if not (
+        isinstance(encoded_text, list) and
+        isinstance(vocabulary, dict) and
+        (isinstance(end_of_word_token, str) or end_of_word_token is None)
+    ):
+        return None
+
+    decoded = ''
+    voc_keys = list(vocabulary.keys())
+    for token in encoded_text:
+        symbol = voc_keys[token]
```

Comment:
So, `voc_keys` is a list... and you are indexing an element in this list... with a token.... no, I don't understand.

Reply:
Tests say ''dict_keys' object is not subscriptable'. That's why I use list()
May be it is not such a good name for a variable, but we have an encoded sequence, and all encoded symbols are compared to identificators in vocabulary which can be compared to indexes of keys in voc_keys. But I can explain it to you tomorrow